src/main/java/frc/robot/subsystems/ElevatorSubsystem.java

```diff
@@ -75,23 +76,31 @@ public double getHeight() {
         + ElevatorConstants.kElevatorLowestHeight);
   }
 
-  // Possible print method for heightGet:
-  /*
-  System.out.println("Encoder: " + m_elevator1Encoder.getPosition() + " rotations");
-  System.out.println(
-      "Elevator Relative Height: "
-          + (m_elevator1Encoder.getPosition() * ElevatorConstants.kElevatorHeightToRot)
-          + " inches");
-  System.out.println(
-      "Elevator Height: "
-          + ((m_elevator1Encoder.getPosition() * ElevatorConstants.kElevatorHeightToRot)
-              + ElevatorConstants.kElevatorLowestHeight)
-          + " inches"); */
+  // Calculated equation based on demo speeds and heights
+  // Sheet used for calculation in software drive
+  public double safetyheight(double x) {
+    return (79.1
+        + 28.4 * x
+        + -725 * Math.pow(x, 2)
+        + 1160 * Math.pow(x, 3)
+        + -536 * Math.pow(x, 4));
+  }
 
   @Override
   public void periodic() {
     // This method will be called once per scheduler run
-    elevatorSpeed =
+    double drivespeed = DrivetrainSubsystem.maxSpeedCmd;
+
+    safetysetpoint = safetyheight(drivespeed);
+    double location;
+    if (safetysetpoint < setpoint) {
```

Comment:
It could be useful to have an indicator in the dashboard that height is being limited for speed so the drive team can get a better feel for the combination of height and speed.